mqtt/mqtt-bridge/src/connectivity_handler.rs

```diff
@@ -0,0 +1,177 @@
+#![allow(dead_code)] // TODO remove when ready
+
+use mqtt3::Event;
+use tokio::sync::mpsc::UnboundedSender;
+use tracing::{debug, info, warn};
+
+use crate::bridge::{BridgeError, BridgeMessage, ConnectivityState};
+
+impl ToString for ConnectivityState {
+    fn to_string(&self) -> String {
+        match self {
+            ConnectivityState::Connected => "Connected".to_string(),
+            ConnectivityState::Disconnected => "Disconnected".to_string(),
+        }
+    }
+}
+
+/// Handles connection and disconnection events and sends a notification when status changes
+pub struct ConnectivityHandler {
+    state: ConnectivityState,
+    sender: UnboundedSender<BridgeMessage>,
+}
+
+impl ConnectivityHandler {
+    pub fn new(sender: UnboundedSender<BridgeMessage>) -> Self {
+        ConnectivityHandler {
+            state: ConnectivityState::Disconnected,
+            sender,
+        }
+    }
+
+    pub async fn handle_connectivity_event(&mut self, event: Event) -> Result<(), BridgeError> {
```

Comment:
```suggestion
    pub fn handle(&mut self, event: Event) -> Result<(), BridgeError> {
```
does it have to be async?

Reply:
I changed to impl EventHandler which is async